### PR TITLE
(SIMP-4206) Added organization provider and updated datasource provider

### PR DESCRIPTION
**Pull request by heliocentric**
- Updated grafana datasource provider to allow datasources to be created
on specific organizations
- Added grafana organization provider to allow organizations to be
created
- Updated README to add more descriptive information for previous
features as well as new

Fixes commercial/enterprise-meta#91
SIMP-4206 #close

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -340,7 +340,25 @@ Example:
 
 The module includes several custom types:
 
-##### `grafana_dashboard`
+#### `grafana_organization`
+
+In order to use the organization resource, add the followingto your manifest:
```
Comment (bastelfreak) [inline README.md]: followingto -> following to

File: lib/puppet/provider/grafana_organization/grafana.rb [hunk 2]
```diff
@@ -0,0 +1,119 @@
+#    Copyright 2015 Mirantis, Inc.
```
Comment (bastelfreak) [inline lib/puppet/provider/grafana_organization/grafana.rb]: 2015?

File: lib/puppet/provider/grafana_organization/grafana.rb [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
```
Comment (bastelfreak) [inline lib/puppet/provider/grafana_organization/grafana.rb]: is that actually needed? The whole project uses the Apache-2 license. We try to keep the header short and simple, and don't add the license text to all files.

**Comment by bastelfreak** [comment]
Hi @heliocentric, thanks for the PR!
* I added a few inline comments, can you take a look at them?
* rubocop isn't happy, can you fix the warnings? Maybe autofix works with bundle exec rake rubocop:auto_correct
* Can you add a few tests for this?

**Comment by brandonrdn** [comment]
@bastelfreak regarding the tests, are you referring to general spec tests or acceptance? I've added a test for the organization provider as well as updating the datasource test. I can add acceptance tests if needed, but it doesn't seem like those are utilized in this module.

**Comment by brandonrdn** [comment]
@bastelfreak With the new push from @heliocentric I believe I've fixed all of your concerns. 

- The Copyright notices have been removed from new files (I copied them over from the other files that were created by bfraser to keep consistency).
- The syntax error has been fixed in the README
- New spec test created for organization provider
- Updated spec  test for datasource provider
- Fixed all issues that rubocop had (0 issues currently)

Please let me know if there is anything else needed to complete this merge.

**Comment by bastelfreak** [comment]
Thanks, I will check the code later. I need to find out why travis isn't working for this PR.

**Comment by bastelfreak** [comment]
github UI isn't showing me any tests, but travis tells me it tested:
https://travis-ci.org/voxpupuli/puppet-grafana/builds/333847788
Can you take a look at the rubocop issue?

**Comment by brandonrdn** [comment]
@bastelfreak which rubocop issue are you referring to? The travis tests you linked are from the original request 4 days ago, doesn't look like they've been re-run.

**Comment by bastelfreak** [comment]
now travis is running again, let's see if it works

**Comment by bastelfreak** [comment]
@heliocentric @brandonrdn now we got issues :)

**Comment by brandonrdn** [comment]
Hey @bastelfreak, I fixed the failures with the spec tests, that was an issue on my end for testing before committing changes and thinking they passed. As of right now, all spec tests pass and rubocop has 0 issues. I'll be pushing up from my fork to the SIMP fork, so you should see the push soon.

**Comment by brandonrdn** [comment]
@bastelfreak The newest updates are pushed. You may need to restart Travis again.

**Comment by brandonrdn** [comment]
So close. I have the fix for the last few rubocop fails. It isn't telling me locally that it's broken.

**Comment by bastelfreak** [comment]
rubocop sitll isn't happy. Can you have a look? you can execute it locally with:

```
bundle exec rake rubocop
```

you can invoke the autofixmagic with:

```
bundle exec rake rubocop:auto_correct
```

you maybe need to install the gems, if you didn't do that:

```
bundle install --path .vendor/ --without development --without system_tests --without release
```

**Comment by brandonrdn** [comment]
@bastelfreak I'm not sure why it's doing this, but I did all 3 previously. I bundle installed, ran the rubocop check, auto_corrected what I could, and manually fixed the rest. When I pushed up my last changes rubocop showed 0 issues. Next push should work. Fingers crossed.

**Comment by bastelfreak** [comment]
now it looks good. Thanks for the work!

**Pull request merged**